crates/oxide/src/lib.rs

```diff
@@ -65,24 +74,89 @@ pub fn clear_cache() {
     cache.clear();
 }
 
-pub fn scan_dir(opts: ScanOptions) -> ScanResult {
-    init_tracing();
+#[derive(Debug, Clone, Default)]
+pub struct Scanner {
+    /// Auto content configuration
+    auto_content: Option<AutoContent>,
+
+    /// Glob sources
+    sources: Option<Vec<GlobEntry>>,
+
+    /// All files that we have to scan
+    files: Vec<PathBuf>,
+
+    /// All generated globs
+    globs: Vec<GlobEntry>,
+
+    /// All found candidates in the files
+    candidates: Vec<String>,
+}
 
-    let (mut files, mut globs) = match opts.base {
-        Some(base) => {
-            // Only enable auto content detection when `base` is provided.
-            let base = Path::new(&base);
-            let (files, dirs) = resolve_files(base);
-            let globs = resolve_globs(base, dirs);
+impl Scanner {
+    pub fn new(auto_content: Option<AutoContent>, sources: Option<Vec<GlobEntry>>) -> Self {
+        init_tracing();
```

Comment:
also thinking maybe this should be not in ::new() either but can't think of a good place to put it. Maybe it should be in `scan`?